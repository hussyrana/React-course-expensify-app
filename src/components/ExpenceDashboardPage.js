import React from 'react';
import ExpenseList from './expenseList';
import ExpenseListFilters from './ExpenseListFilters';
const ExpenceDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);
export default ExpenceDashboardPage;