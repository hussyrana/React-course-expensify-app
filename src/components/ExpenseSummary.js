import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expensesTotal';
import numeral from 'numeral'
export const ExpenseSummary = ({ExpenseCount, ExpenseTotal}) => {
    const expWord = ExpenseCount===1 ? 'expense' : 'expenses';
    const formatedExpenseTotal = numeral(ExpenseTotal/100).format('$0,0.00');
    return(
    <h1>Viewing {ExpenseCount} {expWord} totalling {formatedExpenseTotal}</h1>
    );

};

const mapStateToProps = (state) => {
    
     const expenses = selectExpenses(state.expenses, state.filters)
    
    return{
        ExpenseCount: expenses.length,
        ExpenseTotal: expensesTotal(expenses)

    }
};

export default connect(mapStateToProps)(ExpenseSummary);