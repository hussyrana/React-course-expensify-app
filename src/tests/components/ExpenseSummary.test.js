import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';
test('should render expense summary with 1 expense', ()=>{
    const wrapper = shallow(<ExpenseSummary ExpenseCount = {1} ExpenseTotal={95}/>);
    expect(wrapper).toMatchSnapshot();
});
test('should render expense summary with multiple expenses', ()=>{
    const wrapper = shallow(<ExpenseSummary ExpenseCount = {11} ExpenseTotal={111222295}/>);
    expect(wrapper).toMatchSnapshot();
    
});