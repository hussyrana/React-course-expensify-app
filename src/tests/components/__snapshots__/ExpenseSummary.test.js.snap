// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 1 expense 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling 
  $0.95
</h1>
`;

exports[`should render expense summary with multiple expenses 1`] = `
<h1>
  Viewing 
  11
   
  expenses
   totalling 
  $1,112,222.95
</h1>
`;
